flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -378,31 +390,271 @@ public void registerType(Class<?> type) {
 	// --------------------------------------------------------------------------------------------
 
 	/**
-	 * Creates a DataStream that represents the Strings produced by reading the
-	 * given file line wise. The file will be read with the system's default
-	 * character set.
-	 * 
+	 * Creates a new data stream that contains a sequence of numbers. The data stream will be created with parallelism
+	 * one, so the order of the elements is guaranteed.
+	 *
+	 * @param from
+	 * 		The number to start at (inclusive)
+	 * @param to
+	 * 		The number to stop at (inclusive)
+	 * @return A data stream, containing all number in the [from, to] interval
+	 */
+	public DataStreamSource<Long> generateSequence(long from, long to) {
+		if (from > to) {
+			throw new IllegalArgumentException("Start of sequence must not be greater than the end");
+		}
+		return addSource(new GenSequenceFunction(from, to), "Sequence Source");
+	}
```

Comment:
The `GenSequenceFunction` is a `ParallelSource` itself, this would give the same behaviour as the `generateParallelSource()`.